server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1095,11 +1395,64 @@ public void onFailure(String source, Exception e) {
 
             @Override
             public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
-                deleteSnapshotsFromRepository(repoName, snapshotIds, listener, repositoryStateId, newState.nodes().getMinNodeVersion());
+                addDeleteListener(newDelete.uuid(), listener);
+                if (reusedExistingDelete) {
+                    return;
+                }
+                if (newDelete.state() == SnapshotDeletionsInProgress.State.STARTED) {
+                    if (tryEnterRepoLoop(repoName)) {
+                        deleteSnapshotsFromRepository(newDelete, repositoryData, newState.nodes().getMinNodeVersion());
+                    } else {
+                        logger.trace("Delete [{}] could not execute directly and was queued", newDelete);
```

Comment:
Idk, this is super trivial and I mainly just added it for my debugging of tests. It's just that this is logged in the  corner case of starting a delete in the short window of time between the cluster state update that removed the latest snapshot for a given repo and the listener for that snapshot getting resolved. We can't do synchronisation there because we delete outdated index-N between the two events so this is how I solved it for now, but it's extremely unlikely to get here and completely irrelevant outside of tricky debugging.